docs/pages/access-controls/reference.mdx

```diff
@@ -260,6 +262,10 @@ spec:
 
 ### Extended Node labels syntax
 
+Any string starting with `^` and ending with `$` will be interpreted as a regex and not as a literal string.
+For example, specifying a node label `environment: 'test|staging'` will only match the exact environment string "test|staging",
+while `environment: '^test|staging$'` will match either the string "test" or "staging".
```

Comment:
We should also mention that for strings that do not start with `^` or end with `$`, an `*` can be used as a glob-style matcher that matches any string.